core/src/main/java/com/graphhopper/routing/util/tour/TourStrategy.java

```diff
@@ -0,0 +1,83 @@
+/*
+ *  Licensed to GraphHopper and Peter Karich under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing.util.tour;
+
+import java.util.Random;
+
+/**
+ * Defines the strategy of creating tours.
+ *
+ * @author Robin Boldt
+ */
+public abstract class TourStrategy
+{
+    protected final Random random;
+    protected final double overallDistance;
+
+    public TourStrategy( Random random, double distanceInMeter )
+    {
+        this.random = random;
+        this.overallDistance = distanceInMeter;
+    }
+
+    /**
+     * Defines the number of points that are generated
+     */
+    public abstract int getNumberOfGeneratedPoints();
+
+    /**
+     * Returns the distance in KM that is used for the generated point #iteration
+     */
+    public abstract double getDistanceForIteration( int iteration );
+
+    /**
+     * Returns the bearing between 0 and 360 for the current iteration.
+     * <p>
+     * TODO NOW make identical orientation as the heading
```

Comment:
Here I mean this heading definition https://github.com/graphhopper/graphhopper/blob/master/docs/core/routing.md#heading